### ISSUE TITLE
Refactor

### ISSUE THREAD

**Issue by tigertv** [DanielPetelin/Gordon_Freeman_on_canvas_of_pixels#1]
Привет!
B школе тоже программировал подобным образом. Я только использовал миллиметровку. Брат нарисовал картинку пантеры, а я вырисовывал линии по точкам.

Работа хорошая, но можно улучшить. Что если захочется быстро перенести целую картинку на другие координаты? В данном случае нужно будет менять очень много значений. Чтобы этого не делать необходимо привязываться к переменным. Ну, например:
```
let x = 0;
let y = 0;
ctx.lineTo(x+140, y+20);
```
Eще, думаю, можно сократить количество инструкций использованием [fillRect ](https://developer.mozilla.org/ru/docs/Web/API/CanvasRenderingContext2D/fillRect)

Удачи!

**Comment by adslbarxatov** [comment]
Полностью поддерживаю. А ещё очень хочеться вынести все координаты в отдельный массив, а все lineTo выполнять в цикле. Только не знаю, как в JS циклы работают. Поправьте, если не прав

**Comment by tigertv** [comment]
@adslbarxatov, Да, все верно - данные выносятся из кода обработки и инструкция lineTo не пишется много раз. [Циклы в JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Loops_and_iteration) работают также как и во многих других языках. Получится, что-то вроде:

```javascript
const coords = [
    [140, 20],
    [5, 7]
];
let x = 0;
let y = 0;

for (let item of coords) {
    ctx.lineTo(x + item[0], y + item[1]);
}

```

**Comment by adslbarxatov** [comment]
О, вообще красота будет! Спасибо!

**Comment by tigertv** [comment]
@adslbarxatov , Чтоб еще красивее было, нужно писать комментарии для кода, что он рисует, например голову, ноги и т.д., потому что так сразу не найдешь если захочешь поправить что-то(много строк). Ну или выносить код в функции с осмысленными именами.

**Comment by adslbarxatov** [comment]
Согласен. Но в любом случае координатные данные лучше сначала вытащить из блока исполнения. В инициализации массива их можно и откомментировать. После этого функции сами собой оформятся

**Comment by DanielPetelin** [comment]
Спасибо за советы. Извините, что долго не отвечал, я только недавно научился комментарии читать. В этом проекте придётся долго исправлять если переменные заводить или функцию писать, поэтому этот проект я оставлю так, но может как-нибудь руки дойдут сделаю новый проект с функцией или переменными.
